app/models.py

```diff
@@ -929,6 +931,7 @@ def secret(self, secret):
             self._secret = signer.sign(str(secret))
 
 
+ApiKeyType = Literal["normal", "team", "test"]
 KEY_TYPE_NORMAL = "normal"
 KEY_TYPE_TEAM = "team"
 KEY_TYPE_TEST = "test"
```

Comment:
Should use these constants in the `ApiKeyType`

Reply:
So I tried:
```ApiKeyType  = Literal[KEY_TYPE_NORMAL, KEY_TYPE_TEAM, KEY_TYPE_TEST]```

But that throws an error since a `Literal` needs to be composed of other `Literal`s such as strings, and not pointers. So I thought maybe I could re-assign `KEY_TYPE_NORMAL = Literal["normal"]` and then the statement above would work. But then `KEY_TYPE_NORMAL` just represents a type rather than a value, and can no longer be used like this: `if some_var == KEY_TYPE_NORMAL :...`, since that will throw an error.

So an alternative that would work would be to set:
```
KeyTypeNormal = Literal["normal"]
KEY_TYPE_NORMAL: KeyTypeNormal = "normal"
...
ApiKeyType = Literal[KeyTypeNormal, ...]
```

But that doesn't decrease the amount of repetition needed to create the type. 

So I couldn't get around the duplication, but let me know if you can figure one out.

Reply:
Okay so it turns out `ApiKeyType = Literal[KEY_TYPE_NORMAL, KEY_TYPE_TEAM, KEY_TYPE_TEST]  # type: ignore ` works! The type created is correct, and we can just ignore mypy's complaints about this.